src/frontend/js/utils/api/web-analytics/google_tag_manager.ts

```diff
@@ -0,0 +1,50 @@
+import { WebAnalyticsAPI } from 'types/WebAnalytics';
+
+/**
+ *
+ * Google Tag Manager Richie Web Analytics API Implementation
+ *
+ * This implementation is used when web analytics is configured has `google_tag_manager`.
+ * It will send events to the google tag manager.
+ *
+ */
+const GoogleTagManagerApiInterface = (): WebAnalyticsAPI => {
+  return {
+    type: (): string => {
+      return 'google_tag_manager';
+    },
+    send_enrolled_event: async (resourceLink: string): Promise<any> => {
+      window.dataLayer.push({
+        event: 'courseEnroll',
+        eventProps: {
+          category: 'courseEnroll',
+          action: 'courseEnrollApi',
+          label: resourceLink,
+          // value: value
+        },
+      });
+    },
```

Comment:
Probably in future. I want to send events when a user clicks on enroll button when it is an anchor to an external platform. If I would do this change, I think it would be prefered to also create an abstract implementation. So each implementation `google_analytics.ts` and `google_tag_manager.ts` should only requires to implement that generic `sendEvent` function.
I'm going to try to find this approach on typescript.